test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc021.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.jvmti.tests.redefineClasses;
+
+import com.ibm.jvmti.tests.util.Util;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+public class rc021 {
+	public static native boolean redefineClass(Class name, int classBytesSize,
+			byte[] classBytes);
+
+	static final int ITERATIONS = 100;
+	private static boolean result = true;
+
+	public boolean testAnnotationCacheAfterRedefine() {
+		try {
+			final class AnnotationFetch implements Runnable {
+				@Override
+				public void run() {
+					try {
+						Class original = rc021_testAnnotationCache_O.class;
+						Field annotationCache = Class.class.getDeclaredField("annotationCache");
+						annotationCache.setAccessible(true);
+						annotationCache.set(original, null);
+
+						Annotation[] ann = original.getDeclaredAnnotations();
+						if (ann.length == 0) {
+							System.out.println("No annotations");
+						} else {
+							System.out.println("Annotations:");
+						}
+						for (Annotation a : ann) {
+							System.out.println(a);
+						}
+					} catch (Throwable t) {
+						System.err.println("Exception during annotation fetch:");
+						t.printStackTrace();
+						result = false;
+					}
+				}
+			};
+			final class Redef implements Runnable {
+				@Override
+				public void run() {
+					boolean redefined = Util.redefineClass(rc021.class, rc021_testAnnotationCache_O.class, rc021_testAnnotationCache_R.class);
```

Comment:
Is the intention that this runnable will do something like this
1. transform `rc021_testAnnotationCache_O` to `rc021_testAnnotationCache_Ov2` with `rc021_testAnnotationCache_R` bytes
2.  transform `rc021_testAnnotationCache_O` to `rc021_testAnnotationCache_Ov3` with `rc021_testAnnotationCache_O` bytes
3. repeat

Is that it?

Reply:
Yes, the intention is to alternate between annotation and no annotation in the constant pool

Reply:
Okay,

After the first redefinition

`rc021_testAnnotationCache_O` becomes `rc021_testAnnotationCache_Ov2` with original `rc021_testAnnotationCache_R` bytes

The second redinition does


`rc021_testAnnotationCache_R` to `rc021_testAnnotationCache_Rv2` using `rc021_testAnnotationCache_Ov2` which is original `rc021_testAnnotationCache_R` bytes

So unles, I've missed something at the end of one iteration, original `rc021_testAnnotationCache_O` are lost and both are original `rc021_testAnnotationCache_R` bytes.

Reply:
`Util.redefineClass` calls `Util.loadRedefinedClassBytesWithOriginalClassName(class1, class2);` which gets the class bytes from its .jar  or .class file. When I print `klass.getCanonicalName()` in `getClassBytes` I get 1000 _O and 1000 _R so I think it is using class files to redefine.

Reply:
ah okay, I didn't realize it was loading it from the .jar